### ISSUE TITLE
sst.aws.Nextjs: server.edge.viewerResponse.injection does nothing

### ISSUE THREAD

**Issue by ghost** [anomalyco/sst#5073]
following the [provided example](https://sst.dev/docs/component/aws/nextjs#server-edge-viewerresponse-injection) does not create a cloudfront function.

you can replicate the issue using a blank project as follows:
```
npx create-next-app@14 aws-nextjs
cd aws-nextjs
npx sst@latest init
```
```
# update sst.config.ts
/// <reference path="./.sst/platform/config.d.ts" />

export default $config({
  app(input) {
    return {
      name: "aws-nextjs",
      removal: input?.stage === "production" ? "retain" : "remove",
      home: "aws",
    };
  },
  async run() {
    new sst.aws.Nextjs("MyWeb", {
      server: {
        edge: {
          viewerResponse: {
            injection: `event.response.headers["x-foo"] = "bar";`
          }
        }
      }
    });
  },
});
```
```
npx sst deploy --stage=production
```

**Comment by jayair** [comment]
I'll take a look.

**Comment by leoalves** [comment]
According to the CF functions docs you need an object to set the value of the header. 
https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/example_cloudfront_functions_add_cache_control_header_section.html

Your code should be:

```
export default $config({
  app(input) {
    return {
      name: "aws-nextjs",
      removal: input?.stage === "production" ? "retain" : "remove",
      home: "aws",
    };
  },
  async run() {
    new sst.aws.Nextjs("MyWeb", {
      server: {
        edge: {
          viewerResponse: {
            injection: `event.response.headers["x-foo"] = {value: 'bar'};`
          }
        }
      }
    });
  },
});
```

**Comment by ghost** [comment]
Hi @leoalves,

even though you are right, that's not the issue here.
Using the wrong cloudfront-js should still create a function which then fails on execution.
At least that's what happens when you try to do the same with viewerRequest injection instead.

**Comment by jayair** [comment]
I couldn't replicate this, the function got created for me.

<img width="941" alt="Screenshot 2024-11-25 at 11 56 03 AM" src="https://github.com/user-attachments/assets/b0d2c8d1-4e37-4c54-b1e6-29c6cc84c649">

```
export default $config({
  app(input) {
    return {
      name: "aws-nextjs",
      removal: input?.stage === "production" ? "retain" : "remove",
      home: "aws",
    };
  },
  async run() {
    new sst.aws.Nextjs("MyWeb", {
      server: {
        edge: {
          viewerResponse: {
            injection: `event.response.headers["x-foo"] = {value: "bar"};`
          }
        }
      }
    });
  },
});
```

**Comment by ghost** [comment]
Very odd. Retried with the latest version and still facing this issue. Also tried with different application names. Will try within another aws account and report back.

**Comment by ghost** [comment]
@jayair I think the function in your screenshot is the viewer**Request**-function
I've tried with another aws account and can replicate the issue there as well. no viewer**Response**-function is created and the `x-foo`-response-header is not set

**Comment by jayair** [comment]
It might be a bug. @fwang

**Comment by fwang** [comment]
Fixed in v3.3.41
